messaging/app/src/main/java/com/google/firebase/quickstart/fcm/kotlin/MainActivity.kt

```diff
@@ -84,6 +88,32 @@ class MainActivity : AppCompatActivity() {
         Toast.makeText(this, "See README for setup instructions", Toast.LENGTH_SHORT).show()
     }
 
+    override fun onResume() {
+        super.onResume()
+
+        // [START send_reg_token]
+        FirebaseMessaging.getInstance().token
+            .addOnCompleteListener { task: Task<String> ->
+                if (!task.isSuccessful) {
+                    Log.w(
+                        TAG,
+                        "Fetching FCM registration token failed",
+                        task.exception
+                    )
+                    return@addOnCompleteListener
+                }
+                val token = task.result
+                val tokenMap: MutableMap<String, String?> =
+                    HashMap()
+                tokenMap["fcm_token"] = token
+
+                // Send token to backend server
+                val callable = FirebaseFunctions.getInstance().getHttpsCallable("updateToken")
```

Comment:
Maybe we  can demonstrate KTX here too:
```suggestion
                val callable = Firebase.functions.getHttpsCallable("updateToken")
```